keel-titus-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/titus/cluster/TitusClusterHandler.kt

```diff
@@ -313,6 +320,52 @@ class TitusClusterHandler(
       false
     }
 
+  // todo eb: capture this heuristic and document it for users
+  private fun ResourceDiff<TitusServerGroup>.disableOtherServerGroupJob(resource: Resource<TitusClusterSpec>, desiredVersion: String): Map<String, Any?> {
+    val current = requireNotNull(current) {
+      "Current server group must not be null when generating a disable job"
+    }
+    // todo eb: figure out what server group to disable
+    val existingServerGroups: Map<String, List<ClouddriverTitusServerGroup>> = runBlocking { getExistingServerGroupsByRegion(resource) }
+    val sgInRegion = existingServerGroups.getOrDefault(current.location.region, emptyList()).filterNot { it.disabled }
+
+    if (sgInRegion.size < 2) {
+      log.error("Diff says this is not the only active server group, but now we say otherwise. " +
+        "What is going on? Existing server groups: {}", existingServerGroups)
```

Comment:
So this would maybe be a transient condition due to something else acting on the server group we initially detected. I think the right thing to do in that case is pretty much always to just stop and check again next cycle.